src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidUninstallCommand.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.DotNet.XHarness.Android;
+using Microsoft.DotNet.XHarness.CLI.CommandArguments.Android;
+using Microsoft.DotNet.XHarness.Common.CLI;
+using Microsoft.DotNet.XHarness.Common.CLI.CommandArguments;
+using Microsoft.DotNet.XHarness.Common.CLI.Commands;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.XHarness.CLI.Commands.Android
+{
+    internal class AndroidUninstallCommand : XHarnessCommand
+    {
+        private readonly AndroidUninstallCommandArguments _arguments = new AndroidUninstallCommandArguments();
+
+        protected override XHarnessCommandArguments Arguments => _arguments;
+
+        protected override string CommandUsage { get; } = "android uninstall --package-name=... [OPTIONS]";
+
+        private const string CommandHelp = "Uninstall an .apk from an Android device";
+        protected override string CommandDescription { get; } = @$"
+{CommandHelp}
+ 
+Arguments:
+";
+
+        public AndroidUninstallCommand() : base("uninstall", false, CommandHelp)
+        {
+        }
+
+        protected override Task<ExitCode> InvokeInternal(ILogger logger)
+        {
+            logger.LogDebug($"Android Uninstall command called: App = {_arguments.PackageName}{Environment.NewLine}");
+
+
+            // Package Name is not guaranteed to match file name, so it needs to be mandatory.
+            string apkPackageName = _arguments.PackageName;
+
+            var runner = new AdbRunner(logger);
+
+            try
+            {
+                using (logger.BeginScope("Find device where to uninstall APK"))
+                {
+                    // Make sure the adb server is started
+                    runner.StartAdbServer();
+
+                    if (_arguments.DeviceId == null)
+                    {
+                        return Task.FromResult(ExitCode.ADB_DEVICE_ENUMERATION_FAILURE);
+                    }
+
+                    runner.SetActiveDevice(_arguments.DeviceId);
+
+                    // Wait til at least device(s) are ready
+                    runner.WaitForDevice();
```

Comment:
This could be bad if something goes wrong with the device since it waits for up to 5 minutes by default which will block the uninstall. I think we can remove this and just assume the device is present.

Reply:
Hmmm what error would we get if not present?